features/org.eclipse.equinox.executable.feature/library/gtk/make_linux.mak

```diff
@@ -43,6 +43,20 @@ JAVA_HOME ?= $(shell readlink -f /usr/bin/java | sed "s:jre/::" | sed "s:bin/jav
 PROGRAM_OUTPUT ?= eclipse
 PROGRAM_LIBRARY = $(PROGRAM_OUTPUT)_$(LIB_VERSION).so
 
+# Permitted dependencies and glibc versions
+# This does not include libraries that are dynamically loaded with dlopen,
+# like all the gtk libraries. The purpose of this list, and the version
+# limit on glibc is to ensure eclipse launcher will be able to start
+# on the widest range on Linuxes.
+# All other error handling regarding missing/problematic libraries
+# can be done at runtime.
+PERMITTED_LIBRARIES=libc.so.6 libpthread.so.0 libdl.so.2
+ifeq ($(DEFAULT_OS_ARCH),x86_64)
+PERMITTED_GLIBC_VERSION=2.7
+else
+PERMITTED_GLIBC_VERSION=2.31
```

Comment:
Other archs might haven even higher glibc as we use real machines there, so I wonder if we should possibly just using the latest GLibC version here as long as no one is adding dedicated entries there?

Reply:
OK - I was unsure about what to do on this one. I based this on what I just updated the CDT natives too for CDT 12. I am going to apply this check (for now) to just x86_64. 

BTW for CDT 12 we cross-compile *all* natives from Linux docker container. We have fewer dependencies, in particular no gtk to worry about. The advantage of the cross compile is that we have tooling to analyse all compiled artifacts in a single build.

Reply:
Cross-compiling would indeed reduce requirements on the infrastructure. I know that it's possible for Windows (and how) and already done for Mac. I assume you could tell how to do it on Linux.
I'm not aware of the concerns, but I would be interested to hear them. However I think this should be discussed on a separate issue.

Reply:
I'm not going to raise an issue on this now as realistically no one is likely to action it. The extra dependencies of gtk + ui for the cross compiling makes launcher + swt a more complicated problem to solve that the relatively simple options we have in CDT.

Marking this resolved as I did the key change @laeubi requested in https://github.com/eclipse-equinox/equinox/pull/834#discussion_r1938872428